core/storage/trie/impl/persistent_trie_batch_impl.hpp

```diff
@@ -54,16 +56,16 @@ namespace kagome::storage::trie {
         std::shared_ptr<PolkadotTrie> trie,
         RootChangedEventHandler &&handler);
 
-    void init();
+    // retrieves the current extrinsic index from the storage
+    outcome::result<common::Buffer> getExtrinsicIndex() const;
```

Comment:
declarating a method that gets specific key in a common trie doesn't look right to me

Reply:
Well, it is actually a 'well-known storage key', one of a short list of them, and it makes more sense to put it here, than in the changes tracker which currently needs it, to avoid making the changes tracker depend on a whole batch while all it requires is just one key-value. Since it is a well-known storage key, I assumed it does not violate logic to put it here for now.

Reply:
It's not the only option. You could just use `get(":extrinsic_index"), or smth more smooth. You know the issue, I hope it won't be a common practice to place here "known keys".